client/src/pages/course-content.tsx

```
@@ -0,0 +1,255 @@
+                  <Card key={item.id || index} className="shadow-md hover:shadow-lg transition-shadow">
+                    <CardHeader className="flex flex-row items-center justify-between">
+                      <div className="flex items-center">
+                        <ContentIcon type={item.type || 'lesson'} />
+                        <CardTitle className="text-xl">{item.title}</CardTitle>
+                      </div>
+                      <Badge variant="outline" className="capitalize">{item.type || 'N/A'}</Badge>
+                    </CardHeader>
+                    <CardContent>
+                      {item.description && <p className="text-gray-700 dark:text-gray-300 mb-3">{item.description}</p>}
+                      {item.content && (item.type === 'lesson' || item.type === 'text') && (
+                         <div className="prose dark:prose-invert max-w-none" dangerouslySetInnerHTML={{ __html: item.content }} />
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)